packages/react/src/components/UIShell/HeaderGlobalAction.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import { AriaLabelPropType } from '../../prop-types/AriaPropTypes';
+import Button from '../Button';
+import { usePrefix } from '../../internal/usePrefix';
+
+interface HeaderGlobalActionProps {
+  /**
+   * Required props for the accessibility label of the button
+   */
+  'aria-label'?: string;
+  /**
+   * Required props for the accessibility label of the button
+   */
+  'aria-labelledby'?: string;
+  /**
+   * Provide a custom icon for this global action
+   */
+  children?: ReactNode;
```

Comment:
This one should be required
```suggestion
  children: ReactNode;
```